app/services/account_reset/validate_cancel_token.rb

```diff
@@ -0,0 +1,43 @@
+module AccountReset
+  class ValidateCancelToken
```

Comment:
Would this make more sense under `app/validators/` rather than as a service? Mainly keying off of the `Validate` part of the class name. But I can also see this as a general service or a form. Especially as a form since it returns a `FormResponse`.

Reply:
Agreed. I'll need to refactor cancel now to use the validator